great_expectations/data_context/store/query_store.py

```diff
@@ -87,10 +87,26 @@ def set(self, key, value):
     def get_query_result(self, key, query_parameters=None):
         if query_parameters is None:
             query_parameters = {}
-        query = self._store_backend.get(self._convert_key(key).to_tuple())
+        result = self._store_backend.get(self._convert_key(key).to_tuple())
+        if isinstance(result, dict):
+            query = result.get("query")
+            return_type = result.get("return_type")
```

Comment:
Here's a totally optional suggestion to save some typing for our users in the default case - you could change this line as follows so the return_type can be omitted & default is "list" to be consistent with the documentation on this line: https://github.com/great-expectations/great_expectations/pull/2060/files#diff-48cbe10021881072b0ea56cbc5db449a26790d6c8f18454b9f13df7156d2ad90R31. It'll still raise the error if a user inputs anything other than list or scalar.
```suggestion
            return_type = result.get("return_type", "list")
```

Reply:
Great idea!